storage/aptosdb/src/db/include/aptosdb_reader.rs

```diff
@@ -809,6 +811,87 @@ impl DbReader for AptosDB {
             self.state_store.get_usage(version)
         })
     }
+
+    fn get_db_tailor_iter(
+        &self,
+        start_version: Version,
+        num_transactions: usize,
+    ) -> Result<Box<dyn Iterator<Item = Result<(Transaction, Vec<ContractEvent>)>> + '_,>> {
+        gauged_api("get_db_tailor_iter", || {
+            self.error_if_ledger_pruned("Transaction", start_version)?;
+
+            let txn_iter = self
+                .ledger_db
+                .transaction_db()
+                .get_transaction_iter(start_version, num_transactions)?;
+            let event_vec_iter = self
+            .ledger_db
+            .event_db()
+            .get_events_by_version_iter(start_version, num_transactions)?;
+
+        let zipped = zip_eq(txn_iter, event_vec_iter).enumerate().map(move |(idx, (txn_res, event_vec_res))| {
```

Comment:
zip_eq() panics if one of the iters terminates first. let's avoid that. 

Also, I prefer to zip the txn and event iters on the call site, `get_db_tailor_iter` doesn't seem like a worthy API to be on DbReader.

Reply:
(I almost think we should make txn and event "tailers" independent from each other and run in concurrency from the beginning)

Reply:
changed the zip_eq back to zip. The reason is to have a consistent view between the transaction, events and statevalue index so that a query would see the same value exists in all 3 schemas.